### PR TITLE
P-2587 Data Table Component

### PR DESCRIPTION
**Pull request by gunjankothari**
Snack: https://snack.expo.dev/@gunjankothari/jigsaw---table-component

### CODE REVIEW

File: packages/core/src/components/Table/TablePaginator.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { View, Text } from "react-native";
+
+const TablePaginator = () => (
```
Comment (bluerssen) [inline packages/core/src/components/Table/TablePaginator.tsx:4]: Is this file needed? I dont get what it does.

File: packages/core/src/components/Table/TableHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+);
+
+const styles = StyleSheet.create({
+  wrapper: {
+    display: "flex",
+    flexDirection: "row",
+  },
+});
+
+export default withTheme(TableHeader);
```
Comment (bluerssen) [inline packages/core/src/components/Table/TableHeader.tsx:28]: Lets remove theme and theme objects from these and instead make sure we can just style them with props, and set reasonable defaults on those props.

File: example/src/TableExample.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  },
+  cell: {
+    paddingTop: 16,
+    paddingBottom: 16,
+    paddingLeft: 8,
+    paddingRight: 8,
+  },
+});
+
+export default withTheme(TableExample);
```
Comment (bluerssen) [inline example/src/TableExample.js:207]: Lets remove theme and theme objects from these and instead make sure we can just style them with props, and set reasonable defaults on those props.

File: packages/core/src/utilities.ts [hunk 4]
```diff
@@ -1,5 +1,13 @@
 import { StyleSheet, StyleProp, TextStyle } from "react-native";
-import { isString, isNumber, pick, pickBy, identity } from "lodash";
+import {
+  isString,
+  isNumber,
+  pick,
+  pickBy,
+  identity,
+  omitBy,
```
Comment (bluerssen) [inline packages/core/src/utilities.ts:8]: In the interest of getting rid of things that might be causing the import errors, do we need these new ones?

File: packages/core/src/components/Table/TablePaginator.tsx (hunk 1 above)
Comment (gunjankothari) [reply packages/core/src/components/Table/TablePaginator.tsx:4]: I had created this file initially thinking we might support pagination but we decided to go without pagination that time. I can remove this file.

File: packages/core/src/utilities.ts (hunk 4 above)
Comment (gunjankothari) [reply packages/core/src/utilities.ts:8]: these are just lodash methods. 
Purpose of this is to remove properties which are not defined. 

Do you think this can be causing problem?

File: packages/core/src/components/Table/TableHeader.tsx (hunk 2 above)
Comment (gunjankothari) [reply packages/core/src/components/Table/TableHeader.tsx:28]: removed

File: example/src/TableExample.js (hunk 3 above)
Comment (gunjankothari) [reply example/src/TableExample.js:207]: removed

File: packages/core/src/mappings/Table.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      StylesPanelSections.Background,
+      StylesPanelSections.Borders,
+      StylesPanelSections.Size,
+      StylesPanelSections.MarginsAndPaddings,
+      StylesPanelSections.Position,
+      StylesPanelSections.Effects,
+    ],
+    layout: {
+      fontSize: 14,
+      fontWeight: "500",
```
Comment (bluerssen) [inline packages/core/src/mappings/Table.ts:116]: @adnelson

**Comment by YoussefHenna** [comment]
Closed in favor of https://github.com/draftbit/react-native-jigsaw/pull/598
